ddtrace/_linker.py

```diff
@@ -0,0 +1,43 @@
+import collections
+import weakref
+
+from ddtrace.vendor import attr
+
+from . import compat
+
+
+@attr.s
+class Linker(object):
+
+    _thread_id_to_spans = attr.ib(factory=lambda: collections.defaultdict(weakref.WeakSet), repr=False, init=False)
+
+    # FIXME thread safetyness?
+
+    def link_span(self, span):
+        """Link a span to its running environment.
+
+        Track threads, tasks, etc.
+        """
+        # FIXME: clean _thread_id_to_spans when thread is dead
+        # e.g. check the list of existing threads on retrieval (or here on write?)
+        self._thread_id_to_spans[compat.thread_get_ident()].add(span)
+
+    def get_active_spans_from_thread_id(self, thread_id):
+        """Return all actives spans for a thread."""
+        return {
+            span for span in self._thread_id_to_spans.get(thread_id)
+            if not span.finished
+        }
+
+    def get_active_child_span_from_thread_id(self, thread_id):
+        """Return the latest active span for a thread."""
+        # Key is the parent span id, value is the set of children
+        spans = set(self._thread_id_to_spans.get(thread_id))
+        for span in set(spans):
+            if not span.finished:
+                try:
+                    spans.remove(span._parent)
+                except KeyError:
+                    pass
+
+        return {span for span in spans if not span.finished}
```

Comment:
Reuse the other function and use set difference.

```suggestion
        active_spans = self.get_active_spans_from_thread_id(thread_id)
        return active_spans - {span._parent for span in active_spans}
```